security/custom-gas-token.md

```diff
@@ -0,0 +1,83 @@
+- **Risk Assessment:**
+    - High severity, low likelihood
+    - A hack that steals money from the bridge can end the chain. This is not battle tested code but follows best practices like checks, effects, interactions, has thorough unit and end to end tests as well as follows a similar pattern to Fraxtal’s implementation of custom gas token. Fraxtal has not yet been hacked.
+    - A hack that mints money on L2 without it being collateralized by the rollup should not be possible given a standard ERC20 token is used. This is certainly possible to do with a malicious ERC20 token and this is not possible to detect on chain. This means that social consensus must be used to determine the safety of a rollup using a custom gas token.
+    - Another form of this attack involves using `ether` to mint native asset on L2 when the custom gas token feature is enabled. This would undercollaterize the bridge in terms on the custom gas token and turn the native asset into a proportionate representation of `ether` and the custom gas token. This would break withdrawals.
+- **Mitigations:**
+    - This can be mitigated by using offchain checks on the standardness of the ERC20 token before doing any sort of brand association with the custom gas token chain. These checks cannot be automated, manual inspection must be done to ensure that the token is [specs compliant](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/granite/custom-gas-token.md#properties-of-a-gas-paying-token).
+- **Detection:**
+    - An offchain monitoring service could exist that is able to utilize the APIs on the smart contracts to determine that the bridge is not drained on L1 in a malicious way
+- **Recovery Path(s)**:
+    - It is possible to recollateralize the bridge by directly transferring the ERC20 tokens to the bridge. This would enable users to be able to withdraw again.
+
+### Non Standard ERC20 Usage
+
+- **Description:**
+    - This is related to [Bridge Undercollateralization](https://www.notion.so/Bridge-Undercollateralization-552a985fcbed4d95b468f4b4c5f22ec9?pvs=21) as it is one form that the bridge can become undercollaterized. Really anything is possible with a non standard ERC20
+- **Risk Assessment:**
+    - High severity, low likelihood
+    - Given that we communicate with rollup as a service platforms, this is low probability. Chains that don’t listen to us can do whatever they want in their world but they will never be able to be considered standard.
+- **Mitigations:**
+    - We should not co-market with any custom gas token chain unless we have manually checked that their ERC20 token complies with the specs.
```

Comment:
We'd also need to ensure that the token cannot be upgraded to a non-standard ERC20 in the future, right?

Reply:
We could trust the governance of the token to not upgrade to a non standard ERC20 or enforce an immutable contract. I lean towards trusting the token gov, it would be the same as an exchange integration for the ERC20 token